src/internal_events/wasm/hostcall.rs

```diff
@@ -44,30 +60,31 @@ impl Hostcall {
     }
 }
 
-impl InternalEvent for Hostcall {
+impl InternalEvent for WasmHostcallProgress {
     fn emit_logs(&self) {
-        #[cfg(not(feature = "wasm-timings"))]
-        trace!(
-            state = self.state.as_const_str(),
-            call = self.call,
-            role = self.role.as_const_str(),
-        );
-        #[cfg(feature = "wasm-timings")]
-        {
-            if self.elapsed.as_nanos() == 0 {
-                trace!(
+        match self.state {
+            State::Beginning | State::Cached | State::Completed => event!(
+                tracing::Level::TRACE,
+                {
                     state = self.state.as_const_str(),
                     call = self.call,
                     role = self.role.as_const_str(),
-                );
-            } else {
-                trace!(
+                    elapsed_micros = self.elapsed.as_micros() as u64,
+                },
+                "WASM Hostcall invocation.",
+            ),
+            State::Errored => event!(
+                tracing::Level::ERROR,
```

Comment:
Why `event!` with a hardcoded level instead of the normal level-specific macros?

Reply:
I should use those instead. :) I had no reason.